Source/MLX/Device.swift

```diff
@@ -25,7 +25,7 @@ public enum DeviceType: String {
 /// ### See Also
 /// - <doc:using-streams>
 /// - ``StreamOrDevice``
-public final class Device {
+public final class Device: @unchecked Sendable {
```

Comment:
`let ctx: mlx_device` is an OpaquePointer which isn't sendable, but our use of it is thread safe, so we adopt Sendable like this.